nanoFramework.CoreLibrary/System/SpanByte.cs

```diff
@@ -0,0 +1,172 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// Portions Copyright (c) Microsoft Corporation.  All rights reserved.
+// See LICENSE file in the project root for full license information.
+//
+
+namespace System
+{
+    /// <summary>
+    /// Provides a type- and memory-safe representation of a contiguous region of arbitrary byte array
+    /// </summary>
+    public struct SpanByte
+    {
+        private byte[] _array;
+        private readonly int _start;
+        private readonly int _length;
+
+        /// <summary>
+        /// Creates a new System.SpanByte object over the entirety of a specified array.
+        /// </summary>
+        /// <param name="array">The array from which to create the System.Span object.</param>
+        public SpanByte(byte[] array)
+        {
+            _array = array;
+            _length = array != null ? array.Length : 0;
+            _start = 0;
+        }
+
+        /// <summary>
+        /// Creates a new System.SpanByte object that includes a specified number of elements
+        /// of an array starting at a specified index.
+        /// </summary>
+        /// <param name="array">The source array.</param>
+        /// <param name="start">The index of the first element to include in the new System.Span</param>
+        /// <param name="length">The number of elements to include in the new System.Span</param>
+        /// <exception cref="System.ArgumentOutOfRangeException">
+        /// array is null, but start or length is non-zero. -or- start is outside the bounds
+        /// of the array. -or- start and length exceeds the number of elements in the array.
+        /// </exception>
+        public SpanByte(byte[] array, int start, int length)
+        {
+            if (array != null)
+            {
+                if ((length > array.Length - start) || (start >= array.Length))
```

Comment:
I think it should be `start > array.Length` (this is in case someone is slicing and ends up with zero length slice in the end). Also see https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Span.cs#L78-L82

Reply:
@Ellerbach your take on this please...